mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx

```diff
@@ -0,0 +1,115 @@
+import { View } from '@components'
+import { IconName, IconType, ThemedIcon, ThemedTextV2, ThemedTouchableOpacityV2 } from '@components/themed'
+import { useFeatureFlagContext } from '@contexts/FeatureFlagContext'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { RootState } from '@store'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { ScrollView, Text } from 'react-native'
+import { useSelector } from 'react-redux'
+import { PortfolioParamList } from '../PortfolioNavigator'
+
+export interface ActionButtonsProps {
+  name: string
+  icon: IconName
+  iconType: IconType
+  onPress: () => void
+  testID: string
+  badge?: string | number
+}
+
+export function ActionButtons (): JSX.Element {
+  const { isFeatureAvailable } = useFeatureFlagContext()
+  const navigation = useNavigation<NavigationProp<PortfolioParamList>>()
+  const { futureSwaps } = useSelector((state: RootState) => state.futureSwaps)
+
+  const actions: ActionButtonsProps[] = [
+    {
+      name: translate('components/ActionButtons', 'Send'),
+      icon: 'arrow-up-right',
+      iconType: 'Feather',
+      testID: 'send_balance_button',
+      onPress: () => navigation.navigate('Send')
+    }, {
+      name: translate('components/ActionButtons', 'Receive'),
+      icon: 'arrow-down-left',
+      iconType: 'Feather',
+      testID: 'receive_balance_button',
+      onPress: () => navigation.navigate('Receive')
+    }, {
+      name: translate('components/ActionButtons', 'Swap'),
+      icon: 'repeat',
+      iconType: 'Feather',
+      testID: 'swap_button',
+      onPress: () => navigation.navigate({
+        name: 'CompositeSwap',
+        params: {},
+        merge: true
+      })
+    }, {
+      name: translate('components/ActionButtons', 'Transactions'),
+      icon: 'calendar',
+      testID: 'transaction_button',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('TransactionsScreen')
+    }
+  ]
+
+  if (isFeatureAvailable('future_swap') && futureSwaps.length > 0) {
+    actions.splice(2, 0, {
+      name: translate('components/ActionButtons', 'Future swap'),
+      icon: 'clock',
+      iconType: 'Feather',
+      badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length + 1,
```

Comment:
i think we dont need to +1 right?
```suggestion
      badge: futureSwaps.length > 9 ? '9+' : futureSwaps.length,
```